### PR TITLE
[PM-10361] Remove Group.AccessAll from code

### PR DESCRIPTION
**Pull request by eliykat**
## 🎟️ Tracking

https://bitwarden.atlassian.net/browse/PM-10367

## 📔 Objective

Remove `Group.AccessAll` from code.

Do not drop database columns yet. MSSQL already uses default values in all sprocs (see #4551) so no migrations required there. However, this PR does contain EF migrations to give it a default value so that we can keep it as a shadow property until the next release.

## ⏰ Reminders before review

- Contributor guidelines followed
- All formatters and local linters executed and passed
- Written new unit and / or integration tests where applicable
- Protected functional changes with optionality (feature flags)
- Used internationalization (i18n) for all UI strings
- CI builds passed
- Communicated to DevOps any deployment requirements
- Updated any necessary documentation (Confluence, contributing docs) or informed the documentation team

## 🦮 Reviewer guidelines

- 👍 (`:+1:`) or similar for great changes
- 📝 (`:memo:`) or ℹ️ (`:information_source:`) for notes or general info
- ❓ (`:question:`) for questions
- 🤔 (`:thinking:`) or 💭 (`:thought_balloon:`) for more open inquiry that's not quite a confirmed issue and could potentially benefit from discussion
- 🎨 (`:art:`) for suggestions / improvements
- ❌ (`:x:`) or ⚠️ (`:warning:`) for more significant problems or concerns needing attention
- 🌱 (`:seedling:`) or ♻️ (`:recycle:`) for future improvements or indications of technical debt
- ⛏ (`:pick:`) for minor or nitpick changes

### CODE REVIEW

File: src/Infrastructure.EntityFramework/Vault/Repositories/CipherRepository.cs [hunk 1]
```diff
@@ -544,32 +544,10 @@ private static async Task<int> UpdateCollectionsAsync(DatabaseContext context, G
...
-                                        from cg in cg_g.DefaultIfEmpty()
-                                        where o.Id == organizationId &&
-                                            o.Enabled &&
-                                            ou.Status == OrganizationUserStatusType.Confirmed &&
-                                            (ou.AccessAll || !cu.ReadOnly || g.AccessAll || !cg.ReadOnly)
-                                        select c.Id;
+            availableCollectionsQuery =
+                new CollectionsReadByOrganizationIdUserIdQuery(organizationId.Value, userId.Value)
+                    .Run(context)
+                    .Select(c => c.Id);
```
Comment (eliykat) [inline src/Infrastructure.EntityFramework/Vault/Repositories/CipherRepository.cs:550]: This is the same query logic, but with `AccessAll` removed and in a query class for re-use. We already removed `AccessAll` in all queries but missed this one.

File: util/MySqlMigrations/Migrations/DatabaseContextModelSnapshot.cs [hunk 2]
```diff
@@ -2676,8 +2678,12 @@ protected override void BuildModel(ModelBuilder modelBuilder)
...
             modelBuilder.Entity("Bit.Infrastructure.EntityFramework.SecretsManager.Models.ServiceAccount", b =>
                 {
+                    b.Navigation("ApiKeys");
+
                     b.Navigation("GroupAccessPolicies");
 
+                    b.Navigation("ProjectAccessPolicies");
+
                     b.Navigation("UserAccessPolicies");
                 });
```
Comment (eliykat) [inline util/MySqlMigrations/Migrations/DatabaseContextModelSnapshot.cs:2688]: @rkac-bw this is unrelated to my change, but this is an autogenerated file, and this doesn't appear in the migration itself. Is this OK?

**Comment by eliykat** [comment]
@Thomas-Avery can I please get your help with this? I have an SM test failing [here](https://github.com/bitwarden/server/runs/28627756171#r0s31) even though I've made no change to SM code (aside from a few autogenerated changes in the [database snapshot](https://github.com/bitwarden/server/pull/4614#discussion_r1713074565) - but these tests don't touch the db).

File: util/MySqlMigrations/Migrations/DatabaseContextModelSnapshot.cs (hunk 2 above)
Comment (rkac-bw) [reply util/MySqlMigrations/Migrations/DatabaseContextModelSnapshot.cs:2688]: From what I understand, this file is generated and updated by EF when a migration is added or applied It is  a point-in-time reference of the model to assist in migrations. It should be fine.  @withinfocus Thoughts?

File: src/Infrastructure.EntityFramework/Vault/Repositories/CipherRepository.cs (hunk 1 above)
Comment (rkac-bw) [reply src/Infrastructure.EntityFramework/Vault/Repositories/CipherRepository.cs:549]: lgtm

**Review by r-tome — APPROVED** [review]

**Review by shane-melton — APPROVED** [review]

**Review by rkac-bw — APPROVED** [review]
lgtm

**Pull request merged**